index.js

```diff
@@ -66,53 +84,46 @@ class KoaWebMonetization {
     this.buckets.set(id, balance - price)
   }
 
-  paid ({ price, awaitBalance = false }) {
+  receive () {
     return async (ctx, next) => {
-      const id = ctx.params.id
-      if (!id) {
-        return ctx.throw(400, 'ctx.params.id must be defined')
-      }
-
-      const _price = (typeof price === 'function')
-        ? Number(price(ctx))
-        : Number(price)
-
-      if (awaitBalance) {
-        await this.awaitBalance(id, _price)
-      }
-
-      try {
-        this.spend(id, _price)
-        return next()
-      } catch (e) {
-        return ctx.throw(402, e.message)
+      await this.connect()
+      const re = pathToRegexp(this.receiverEndpointUrl)
+      const isMonetizer = re.exec(ctx.request.url)
+      if (ctx.get('Accept').indexOf('application/spsp+json') !== -1 && isMonetizer) {
+        const { destinationAccount, sharedSecret } =
+          this.receiver.generateAddressAndSecret()
+
+        const segments = destinationAccount.split('.')
+        const resultAccount = segments.slice(0, -2).join('.') +
+          '.' + isMonetizer[1] +
+          '.' + segments.slice(-2).join('.')
+
+        ctx.set('Content-Type', 'application/spsp+json')
+        ctx.body = {
+          destination_account: resultAccount,
+          shared_secret: sharedSecret.toString('base64')
+        }
       }
     }
   }
 
-  receiver () {
-    return async ctx => {
-      await this.connect()
-
-      if (ctx.get('Accept').indexOf('application/spsp+json') === -1) {
-        return ctx.throw(404)
-      }
-
-      const { destinationAccount, sharedSecret } =
-        this.receiver.generateAddressAndSecret()
-
-      const segments = destinationAccount.split('.')
-      const resultAccount = segments.slice(0, -2).join('.') +
-        '.' + ctx.params.id +
-        '.' + segments.slice(-2).join('.')
-
-      ctx.set('Content-Type', 'application/spsp+json')
-      ctx.body = {
-        destination_account: resultAccount,
-        shared_secret: sharedSecret.toString('base64')
-      }
+  mount () {
+    return async (ctx, next) => {
+      ;['awaitBalance', 'spend'].forEach(key => {
+        ctx.response[key] = ctx[key] = (amount) => {
+          return this[key](ctx.cookies.get(this.cookieName), amount)
+        }
+        ctx[key] = ctx[key].bind(this)
+      })
+      ctx.cookies.set(this.cookieName, this.generatePayerId(ctx), this.cookieOptions)
+      this.receive()(ctx, next)
+      return next()
     }
   }
+
+  serveClient () {
+    return fs.readFileSync(path.resolve(__dirname, 'client.js'))
```

Comment:
better to make this an async call to `readFile` so the IO doesn't block the rest of the server. The `fs` module's version of readFile only takes a callback but if you use `fs-extra` you can use async/await on it